pkg/kubelet/kubelet_pods.go

```diff
@@ -2105,6 +2105,13 @@ func (kl *Kubelet) convertToAPIContainerStatuses(pod *v1.Pod, podStatus *kubecon
 			} else {
 				preserveOldResourcesValue(v1.ResourceCPU, oldStatus.Resources.Requests, resources.Requests)
 			}
+			// TODO(tallclair,vinaykul,InPlacePodVerticalScaling): Investigate defaulting to actuated resources instead of allocated resources above
+			if _, exists := resources.Requests[v1.ResourceMemory]; exists {
+				// Get memory requests from actuated resources
+				if actuatedResources, found := kl.allocationManager.GetActuatedResources(pod.UID, allocatedContainer.Name); found {
+					resources.Requests[v1.ResourceMemory] = actuatedResources.Requests.Memory().DeepCopy()
```

Comment:
nit: It's already deep-copied when `GetActuatedResources` is called, so don't need to copy here.

I think we've probably gone overboard with the defensive deep copies.

Reply:
And I'm responsible for half of the unnecessary deep copies :( 
I'll fix it shortly!